common: Add pooled video frame allocator

LindaYu17 opened the pull request:
Signed-off-by: Linda Yu <linda.yu@intel.com>

LindaYu17 committed:
common: Add pooled video frame allocator

Signed-off-by: Linda Yu <linda.yu@intel.com>

xuguangxin commented on common/videopool.h:
```diff
@@ -25,11 +25,12 @@ template <class T>
 public:
-    static SharedPtr<VideoPool<T> >
-    create(std::deque<SharedPtr<T> >& buffers)
+    VideoPool(std::deque<SharedPtr<T> >& buffers)
```
Please split this to other commit

xuguangxin commented on common/Makefile.am:
```diff
@@ -3,10 +3,12 @@ libyami_common_source_c = \
 	utils.cpp \
 	nalreader.cpp \
 	surfacepool.cpp \
+	pooledframeallocator.cpp \
```
Please use PooledFrameAllocator.cpp. We should follow webkit style https://webkit.org/code-style-guidelines/ for new file

xuguangxin commented on common/pooledframeallocator.h:
```diff
@@ -0,0 +1,50 @@
+ */
+
+#ifndef POOLED_FRAME_ALLOCATOR_H_
+#define POOLED_FRAME_ALLOCATOR_H_
```
use PooledFrameAllocator_h, it's webkit style

LindaYu17 committed:
common: public VideoPool constructor

Remove create() function as we could not pass deleter to this function
when creating shared_ptr

Signed-off-by: Linda Yu <linda.yu@intel.com>

LindaYu17 commented:
change done @xuguangxin

xuguangxin commented on common/PooledFrameAllocator.cpp:
```diff
@@ -0,0 +1,88 @@
+#include "config.h"
+#endif
+
+#include <PooledFrameAllocator.h>
```
Sorry I missed this, This should be "" instead of <>

LindaYu17 commented:
done @xuguangxin

chivakker commented on common/PooledFrameAllocator.cpp:
```diff
@@ -0,0 +1,88 @@
+    std::deque<SharedPtr<VideoFrame> > buffers;
+    for (size_t i = 0;  i < surfaces.size(); i++) {
+        SharedPtr<VideoFrame> f(new VideoFrame);
+        memset(f.get(), 0, sizeof(VideoFrame));
```
This memset can be moved to the constructor of VideoFrame.  There's a similar problem with the MiscBuffer that is fixed in that way.

xuguangxin replied:
VideoFrame do not have constructor. It is a pod type; it will be use in capi too.

xuguangxin replied:
Could give more detail description for MiscBuffer thing?

chivakker reviewed (approved):
I only have one comment but it minor at this point.  If submitter sees more benefit on pushing the memset to the constructor please do.

xuguangxin commented:
merged as 9c7bb50639233a505599a17d74caa4a0dd0ee0ad
